### DISCUSSION TITLE
Call for input: error mitigation benchmarking GUI (UW capstone project)

### DISCUSSION THREAD

**Discussion by ecarlander** [unitaryfund/mitiq#2340] [Ideas]
Hello! 

My name is Ella, I am working with a couple of fellow UW graduate students on a pipeline to characterize performance vs. overhead for different error mitigation techniques using Mitiq's implementations. This has included the development of a GUI for users to easily compare mitigation techniques. This is done through plotting various measures of overhead and performance (such as # of additional gates or unique circuits required). 

We are hoping to get input from the Unitary Fund community on what features would make this GUI most useful. I've included a screenshot of it's current state (extremely early stage), to provide some context. Please let us know what ideas you may have!! Any and all suggestions are welcome, and please feel free to ask questions.

![image](https://github.com/unitaryfund/mitiq/assets/129551722/ea1b0096-6011-4ad0-90cd-5390a025b0b6)

**Comment by jordandsullivan** [comment]
Hi Ella! This looks very cool ☺️
I know it's in an early stage, but whenever you've got some code to share, maybe you could share it as a GitHub repo? May be helpful to give folks context -- doesn't have to be perfect :)

If you're part of the Unitary Fund discord, we can share this issue with the #mitiq channel to get some more eyes on it. 

Couple questions come to mind for me: 
1. Who are you anticipating will be the primary users of the GUI (e.g. quantum computing researchers, educators, students, etc.)
2. Are you looking for direct user testing or more general feedback?

**Reply by ecarlander** [reply]
Hey Jordan, thanks for the reply :) 

If you didn't already see, Nathan provided a link to our repo in this discussion thread. As for your questions, right now the primary users we have in mind are quantum computing researchers who may want some info/guidance in choosing which error mitigation technique to use. That being said, I think the GUI specifically could be a useful education tool. Currently we are seeking general feedback, but in the future user testing would be very helpful!  

I am part of the UF discord (e11ac), my group and I are signed up to talk about this project on the 5/29 Quantum Wednesday.

**Reply by jordandsullivan** [reply]
Thanks, I did see that! Looking forward to your talk on the 29th :]

**Comment by nathanshammah** [comment]
The code is now available at https://github.com/AlexandrosPeltekis/ErrorMit. I suggest holding a live chat for a presentation of the work!

**Reply by Misty-W** [reply]
Looking forward to it!

**Comment by natestemen** [comment]
Going to close this for now since the GUI isn't being actively worked on. Happy to reopen if it is still something of interest.
